### PR TITLE
Add backend code to get token - iOS

### PR DESCRIPTION
**Pull request by cristobal-buenrostro**

### CODE REVIEW

**Review by rwallerms — CHANGES_REQUESTED** [review]

File: js/samples/ios/README.md [hunk 1]
```diff
@@ -19,6 +19,17 @@
       SUBDOMAIN=<YOUR_SUBDOMAIN>
       ```
 
+1. In case you want to have the iOS sample getting the token from the back end code.Cd into the **server** folder and then [refer to public node server sample here](https://github.com/microsoft/immersive-reader-sdk/tree/dev/js/samples/quickstart-nodejs).
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/ios/README.md:22]: revised

File: js/samples/ios/quickstart-swift/LaunchViewController.swift [hunk 2]
```diff
@@ -169,15 +198,19 @@ class LaunchViewController: UIViewController {
     ///     -theError: The error that occured when the token fails to be obtained from the Azure Active Directory Authentication.
     func getToken(onSuccess: @escaping (_ theToken: String) -> Void, onFailure: @escaping ( _ theError: String) -> Void) {
         let tokenForm = "grant_type=client_credentials&resource=https://cognitiveservices.azure.com/&client_id=" + self.clientId! + "&client_secret=" + self.clientSecret!
-        let tokenUrl = "https://login.windows.net/" + self.tenantId! + "/oauth2/token"
+        let tokenUrl = self.isTokenFromServer ? self.tokenServerUrl! : "https://login.windows.net/" + self.tenantId! + "/oauth2/token"
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/ios/quickstart-swift/LaunchViewController.swift]: new var added

File: js/samples/ios/quickstart-swift/LaunchViewController.swift [hunk 3]
```diff
@@ -5,7 +5,9 @@ class LaunchViewController: UIViewController {
     private var clientId = ProcessInfo.processInfo.environment["CLIENT_ID"]
     private var clientSecret = ProcessInfo.processInfo.environment["CLIENT_SECRET"]
     private var subdomain = ProcessInfo.processInfo.environment["SUBDOMAIN"]
+    private var tokenServerUrl = ProcessInfo.processInfo.environment["TOKEN_SERVER_URL"]
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/ios/quickstart-swift/LaunchViewController.swift]: yeah, makes sense, value removed from config and handled now as variable

File: js/samples/ios/quickstart-swift/LaunchViewController.swift [hunk 4]
```diff
@@ -169,15 +198,19 @@ class LaunchViewController: UIViewController {
...
+        let tokenUrl = self.isTokenFromServer ? self.tokenServerUrl! : "https://login.windows.net/" + self.tenantId! + "/oauth2/token"
 
         var responseTokenString: String = "0"
 
         let url = URL(string: tokenUrl)!
         var request = URLRequest(url: url)
-        request.httpBody = tokenForm.data(using: .utf8)
-        request.httpMethod = "POST"
-
+        if(!self.isTokenFromServer){
```
[the remark this answers is not in the document]
Comment (cristobal-buenrostro) [reply js/samples/ios/quickstart-swift/LaunchViewController.swift]: agree, changed

**Review by rwallerms — APPROVED** [review]

**Pull request merged**